compiler/rustc_lint/src/builtin.rs

```diff
@@ -242,7 +248,8 @@ declare_lint! {
     /// behavior.
     UNSAFE_CODE,
     Allow,
-    "usage of `unsafe` code and other potentially unsound constructs"
+    "usage of `unsafe` code and other potentially unsound constructs",
+    [loadbearing: true]
```

Comment:
Suspicion: are we sure the load-bearing lints are not being skipped? What I'm thinking of is the following scenario, consider:

1. We have a load-bearing lint called `LINT_LOAD_BEARING`.
2. We have an allow-by-default lint called `LINT_ABD` who's not considered load-bearing, and have no other source or cli attributes/flags that would've caused it to be otherwise only `allow`, i.e. so "skippable" in the current will-emit scheme.

But in the implementation of these lints, they are intertwined. `LINT_LOAD_BEARING` is not implemented in a separate lint pass, but instead emitted manually with some `emit_diagnostic()` or whatever, which is only reachable from *within* the lint pass of `LINT_LOAD_BEARING`. Now, I don't think this is a great coupling, but that's a separate discussion.

In this scenario, if `LINT_ABD` is skipped under the current scheme (which you would normally have no other reason to suspect that it can't be skipped), then `LINT_LOAD_BEARING` would now no longer be emitted even if it would be deny-by-default or a hard error, since the analysis leading to the emission of `LINT_LOAD_BEARING` would now be too skipped.

(This is only a suspicion, I have not tried to actually debug this myself)

Reply:
Am I understanding correctly? I think you're saying that if a lint pass contains a manually-emitted diagnostic but is only executed through a non-loadbearing normal lint, the manually-emitted diagnostic could not get executed.

I tried accounting for this (that's the reason some lints are marked as loadbearing, and the filter function checks first for the `loadbearing' field), but I didn't think that non-loadbearing lints could trigger loadbearing diagnostics. We should make those lints loadbearing then.

I'll go check, thanks for the idea!